paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -518,9 +519,31 @@ public void rollbackTo(String tagName) {
             // earliest hint
             SnapshotManager snapshotManager = snapshotManager();
             if (!snapshotManager.snapshotExists(taggedSnapshot.id())) {
+                // clear the changelog manifest when the changelog not exists
+                Snapshot restored = taggedSnapshot;
+                if (!changelogExists) {
+                    restored =
+                            new Snapshot(
+                                    taggedSnapshot.version(),
+                                    taggedSnapshot.id(),
+                                    taggedSnapshot.schemaId(),
+                                    taggedSnapshot.baseManifestList(),
+                                    taggedSnapshot.deltaManifestList(),
+                                    null,
```

Comment:
we need to 'mute' the `changelogManifestList` here, if the changelog and snapshot already not there. Otherwise, the scanner may read on a non-exist manifest list. WDYT ? @JingsongLi

Reply:
![image](https://github.com/apache/paimon/assets/9486140/1ce9aa0d-220a-4ac1-866c-717938658002)

It can be reproduce by disable this block and rerun  `testRollbackWithChangelog`

Reply:
I see `testRollbackWithChangelog`, but rollbacked table has no changelog, why you want to read changelog?

Reply:
Maybe it is a separate issue which is unrelated to changelog decouple.

Reply:
After discuss with @JingsongLi , it's a separate issue, we could solve it in another pr. So revert this change.